pkg/volume/iscsi/iscsi.go

```diff
@@ -177,12 +177,49 @@ func (plugin *iscsiPlugin) newUnmounterInternal(volName string, podUID types.UID
 }
 
 func (plugin *iscsiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	// Find globalPDPath from pod volume directory(mountPath)
+	var globalPDPath string
+	mounter := plugin.host.GetMounter(plugin.GetPluginName())
+	paths, err := mount.GetMountRefs(mounter, mountPath)
+	if err != nil {
+		return nil, err
+	}
+	for _, path := range paths {
+		if strings.Contains(path, plugin.host.GetPluginDir(iscsiPluginName)) {
+			globalPDPath = path
+		}
+	}
+	// Couldn't fetch globalPDPath. Retrun dummy iscsiVolume
```

Comment:
what is the possible cause for not getting the globalPDPath?

Reply:
@jingxu97 If the globalPDPath is not mounted, GetMounter can't get the path from mount list.
This means if the kubelet node is unexpectedly rebooted, we can't find corresponding globalPDPath mount point from mountPath, IIUC.

example of  globalPDPath:
/var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/192.168.122.85:3260-iqn.2017-05.com.example:rhel7-lun-0

Reply:
If kubelet service just temporarily restarted, both mountPath and globalPDPath are still there after restarting kubelet.
Therefore, GetMounter on above can obtain globalPDPath properly.